### ISSUE TITLE
Music library folder was reset to default

### ISSUE THREAD

**Issue by cloo** [YunoHost-Apps/navidrome_ynh#75]
I'm not sure how to check when it happened, but my guess would be it was on last update (probably months ago). 
Using 0.47.5~ynh2 on YunoHost 4.3.5

(Sorry for not using template - it should be clear enough though.. )

**Comment by cloo** [comment]
Also, after setting the library folder back, I found my playlists wiped. All playlists are listed in UI, but with no content. 
When playlist downloaded via UI, the only line in .m3u file is 
`#EXTM3U`

**Comment by ericgaspar** [comment]
it is hard to tell without logs...

**Comment by zecamigo** [comment]
@cloo 

Were you using a custom location for you music library?

@ericgaspar 
This just happened to me because when you upgrade the configuration file is reset.

Which means the music library location is reset to default and immediately after upgrade a scan is performed.  This breaks all the favorites, playlists etc...

I couldn't find a way to stop navidrome from either resetting the configuration file or changing at least not start the service immediately so I can change the library location to my custom setting.

**Comment by ericgaspar** [comment]
The configuration file has been regenerated to be synchronized with the new upstream settings. Some parameters have been added to the config_panel.

**Comment by zecamigo** [comment]
The webadmin upgrades are awesome btw

So the workaround here would be to backup the manually and the restore it after the upgrade?

**Comment by ericgaspar** [comment]
There is a sum check done to the replaced config file to check if it as custom changes, if so a copy is then made beforethe file gets regenerated (I can't recall where this copy is stored...)

**Comment by zecamigo** [comment]
Yes the config file is saved, but before you have time to restore it, navidrome is already scanning the music library and breaking the favorites, playlists because it can't find the music files.

**Comment by zecamigo** [comment]
> So the workaround here would be to backup the database manually and the restore it after the upgrade?

@ericgaspar would this be a viable workaround?
The custom config is saved, it's the scan after the update that instantly resets any favorites, playlists etc based on the files on a custom location.

**Comment by IsidoreIsou** [comment]
Anyone found any solution to that problem?

**Comment by zecamigo** [comment]
@IsidoreIsou @ericgaspar 
This is being discussed here: 
https://github.com/navidrome/navidrome/discussions/2671

And an upstream improvement is being considered.

As this behavior in the YNH package breaks irreversibly the favorites playlist etc  it would it be possible to also fix it also on this side?

**Comment by IsidoreIsou** [comment]
thanks @zecamigo.

Something I'm not sure to understand:

Before the update, is there a location where Navidrome auto backup some config files like the "Recently Added" list? And can I restore this file after the update?

**Comment by zecamigo** [comment]
> Before the update, is there a location where Navidrome auto backup some config files like the "Recently Added" list? 

I'm guessing the date each file is added should be on the database before update.

The only backups should be the automatic ones if you have them running, and the backup ynh does just before the update. I don't know how to extract the database from that backup. 

But even then the database structure might change from one version to the other, so it's not advisable to restore a previous' version database into the upgraded version.

**Comment by zecamigo** [comment]
This issue seems to have been fixed in the upstream code in version 0.51.1 
@cloo @IsidoreIsou
